helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -220,17 +221,24 @@ public GenericHelixController(String clusterName, Set<Pipeline.Type> enabledPipe
   class RebalanceTask extends TimerTask {
     final HelixManager _manager;
     final ClusterEventType _clusterEventType;
+    private final boolean _shouldRefreshCache;
```

Comment:
nit: set default value, if not, Java assumes it's false by default

Reply:
I tried to change it to optional object. So we have strict backward compatibility.